### PR TITLE
Using OSRM for routing over pgrouting

### PR DESCRIPTION
**Pull request by wittjosiah**

### CODE REVIEW

File: api/conf/application.conf [hunk 1]
```diff
@@ -18,3 +18,6 @@ play.evolutions.db.default.useLocks = true
 
 ratings.tileserver.url = "http://localhost:8080"
 ratings.tileserver.url = ${?RATINGS_TILESERVER_URL}
+
+osrm.routing.uri = ${OSRM_ROUTING_URI}
+osrm.nearest.uri = ${OSRM_NEAREST_URI}
```
Comment (kiambogo) [inline api/conf/application.conf]: Are these different enough to merit two separate lines? Why not have osrm.uri and then just use constants for the routing path vs the nearest path? or are the two very different?

File: api/app/services/RoutingService.scala [hunk 2]
```diff
@@ -7,34 +7,44 @@ import com.trifectalabs.roadquality.v0.models.{ Point, MapRoute }
...
-    } map { b: Seq[MapRoute] =>
-      val pl = Polyline.encode(b.flatMap(p => Polyline.decode(p.polyline)).toList)
-    MapRoute(polyline = pl, distance = b.map(_.distance).sum ) }
+class RoutingServiceImpl @Inject()(configuration: Configuration, ws: WSClient)(implicit ec: ExecutionContext) extends RoutingService {
+  lazy val osrmRoutingUri = configuration.getString("osrm.routing.uri").get
+  lazy val osrmNearestUri = configuration.getString("osrm.nearest.uri").get
+
+  def generateRoute(points: String): Future[MapRoute] = {
+    val osrmUrl = s"$osrmRoutingUri/$points"
+    ws.url(osrmUrl).get().map { res =>
```
Comment (kiambogo) [inline api/app/services/RoutingService.scala]: what happens if this fails? we need to handle the case that OSRM is down - i.e., return a 503 and handle gracefully on the client

Comment (kiambogo) [reply api/app/services/RoutingService.scala]: you can pattern match on the status

**Review by kiambogo — CHANGES_REQUESTED** [review]
I also would like the profile stuff to be under resources/orsm/profiles/... personally

**Comment by wittjosiah** [comment]
addressed your comments @kiambogo

File: api/app/services/RoutingService.scala [hunk 3]
```diff
@@ -1,40 +1,49 @@
...
   }
 
   def snapPoint(point: Point): Future[Point] = {
-    mapDao.snapPoint(point)
+    val pointString = s"${point.lng},${point.lat}"
+    val osrmUrl = s"$osrmUri/nearest/v1/cycling/$pointString"
+    ws.url(osrmUrl).withRequestTimeout(5000.millis).get().map { res =>
+      val waypoints = (res.json \ "waypoints").as[List[JsValue]]
+      val location = (waypoints(0) \ "location").as[List[Double]]
+      Point(
```
Comment (kiambogo) [inline api/app/services/RoutingService.scala:42]: This isn't elm - lng should be on the same line as Point. ;)

**Review by kiambogo — APPROVED** [review]

**Pull request merged**